common/membership/interfaces.go

```diff
@@ -82,6 +82,9 @@ type (
 	ServiceResolver interface {
 		// Lookup looks up the host that currently owns the resource identified by the given key.
 		Lookup(key string) (HostInfo, error)
+		// LookupN looks n hosts that owns the resource identified by the given key, if n greater than total number
+		// of hosts total number of hosts will be returned
```

Comment:
I wish ringpop could return exactly n results here (with duplication if necessary)... I wonder if it's worth implementing that extension ourselves

Reply:
@dnr thank you for the review. What you suggest is to return the same HostInfo more then once if n is greater then len of []HostInfo returned by ringpop? Do I understand it right. Something like:
```
result := ringpop.LookN(key, n)
repeat := ceiling(n / result)
result := RepeatSlice(result, repeat)[:n]
```

Reply:
done